src/fastcs/transport/graphQL/graphQL.py

```diff
@@ -8,26 +8,28 @@
 from strawberry.types.field import StrawberryField
 
 from fastcs.attributes import AttrR, AttrRW, AttrW, T
-from fastcs.controller import BaseController, Controller
+from fastcs.controller import BaseController, Controller, _get_single_mapping
 
 from .options import GraphQLServerOptions
 
 
 class GraphQLServer:
     def __init__(self, controller: Controller):
         self._controller = controller
-        self._fields_tree: FieldTree = FieldTree("")
         self._app = self._create_app()
 
     def _create_app(self) -> GraphQL:
-        _add_attribute_operations(self._fields_tree, self._controller)
-        _add_command_mutations(self._fields_tree, self._controller)
-
-        schema_kwargs = {}
-        for key in ["query", "mutation"]:
-            if s_type := self._fields_tree.create_strawberry_type(key):
-                schema_kwargs[key] = s_type
-        schema = strawberry.Schema(**schema_kwargs)  # type: ignore
+        api = GraphQLAPI(self._controller)
+
+        if not api.queries:
+            raise ValueError(
```

Comment:
Fair enough `Every GraphQL schema must support query operations.` - https://graphql.org/learn/schema/#the-query-mutation-and-subscription-types

Reply:
The way you have done it is good - it means we only need one query somewhere in the tree and it will bubble to the top level for the `raise`. I do think though that errors we choose to raise should be possible to handle with `except FastCSException` via the baseclass

Reply:
Sure, happy to use a FastCSException here.